pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -382,21 +386,48 @@ type derivedInput struct {
 	Values      map[string]any
 }
 
-func (d *DeploymentController) render(templateName string, mi TemplateInput) ([]string, error) {
+func (d *DeploymentController) render(templateName string, mi TemplateInput, deployment *appsv1.Deployment) ([]string, error) {
 	cfg := d.injectConfig()
 
 	template := cfg.Templates[templateName]
 	if template == nil {
 		return nil, fmt.Errorf("no %q template defined", templateName)
 	}
+
+	// Default labels are required to properly match a proxy config with a gateway in case where the proxy config was created
+	// before the gateway. This is because in such a case, the gateway deployment would be nil and newly created gateway
+	// would not be matched with the proxy config.
+	podLabels := map[string]string{
+		"istio.io/gateway-name":           mi.Name,
```

Comment:
I think we want _only_ this one, per https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api/#resource-attachment-and-scaling it is the only valid one. That is also known without looking at Deployment which makes it simpler and doesn't have the issues of circular dependency

In the near future it will be 

```yaml
parentRef:
  kind: Gateway
  name: my-gw
```

anyways so this will be short lived

Reply:
I included `"service.istio.io/canonical-name"`, because I wanted to note that if we don't support this label, it may cause regression if someone relied on it. But I don't have strong opinion on that, so I'm removing it as you requested.

Reply:
Using istio.io/gateway-name will also break in the future (probably 1.19, maybe 1.20) anyways - that is why its marked experimental.

Reply:
> Done

Actually, it wasn't done. I forgot to remove pod template labels. But now it's done.